.github/workflows/SharedKernel-ci-cd.yml

```diff
@@ -0,0 +1,118 @@
+    name: Continuous Integration
+    runs-on: ubuntu-latest
+    outputs:
+      latest_version: ${{ steps.tag_generator.outputs.new_version }}
+    steps:
+      - name: Setup .NET Core
+        id: setup_dotnet_core
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 5.0.102
+
+      - name: Checkout repository
+        id: checkout_repo
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Filter changes
+        id: filter_changes
+        uses: dorny/paths-filter@v2
+        with:
+          filters: |
+            is_change_in_project_source:
+              - 'SharedKernel/src/**'
+
+      - if: steps.filter_changes.outputs.is_change_in_project_source == 'true'
+        name: Bump GH tag
+        id: tag_generator
+        uses: mathieudutour/github-tag-action@v5.4
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          default_bump: false
+          release_branches: ${{ steps.data_gatherer.outputs.default_branch }}
+
+      - name: Build solution
+        id: build_solution
+        shell: pwsh
+        run: |
+          dotnet build .\SharedKernel\SharedKernel.sln --configuration Release
+
+      - name: Run unit tests
+        id: run_unit_tests
+        shell: pwsh
+        run: |
+          dotnet test .\SharedKernel\SharedKernel.sln --configuration Release --no-build
+
+      - if: steps.tag_generator.outputs.new_version != ''
+        name: Create NuGet Package
+        id: create_nuget_package
+        shell: pwsh
+        run: |
+          nuget pack .\SharedKernel\src\PluralsightDdd.SharedKernel\PluralsightDdd.SharedKernel.csproj `
+          -OutputDirectory Artifacts `
+          -properties Configuration=Release `
+          -Version ${{ steps.tag_generator.outputs.new_version }}
+
+      - if: steps.tag_generator.outputs.new_version != ''
+        name: Upload artifacts
+        id: upload_artifacts
+        uses: actions/upload-artifact@v1
+        with:
+          name: PluralsightDdd.SharedKernel.${{ steps.tag_generator.outputs.new_version }}
+          path: Artifacts/
+
+  cd:
+    if: needs.ci.outputs.latest_version != ''
+    name: Continuous Deployment
+    needs: ci
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup dotnet
+        id: setup_dotnet
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 5.0.102
+
+      - name: Checkout repository
+        id: checkout_repo
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Download artifacts
+        id: download_artifacts
+        uses: actions/download-artifact@v2
+        with:
+          name: PluralsightDdd.SharedKernel.${{ needs.ci.outputs.latest_version }}
+          path: Artifacts/
+
+      - name: Publish to NuGet
+        id: publish_to_nuget
+        shell: pwsh
+        run: |
+          nuget push ./Artifacts/PluralsightDdd.SharedKernel.${{ needs.ci.outputs.latest_version }}.nupkg `
```

Comment:
It is important not to change package naming schemes that the author of the project has chosen. Check out his publish pipeline and follow the naming scheme of the package in our pipeline as well.

NOTE: This applies for the creation of the package.

Reply:
Checked and everything is the same as the owner's published package.